src/Jackalope/Property.php

```diff
@@ -709,25 +685,44 @@ public function _setValue($value, $type = PropertyType::UNDEFINED)
          */
 
         if (PropertyType::UNDEFINED === $type) {
-            $type = $this->valueConverter->determineType(is_array($value) ? reset($value) : $value);
+            if (is_array($value) && 0 === count($value)) {
+                $type = PropertyType::STRING;
```

Comment:
ah, i know why i did not: this is in the phpcr-utils, further complicating everything.

i just wanted to fix an edge case, not refactor everything :-( will do a PR on phpcr-utils and bump its version